### DISCUSSION TITLE
Question about program NoiseAccelerometer

### DISCUSSION THREAD

**Discussion by Kyrie714** [groops-devs/groops#254] [Q&A] [answered]
Hello, developers!

I'm trying to add noise to simulated accelerometer data through NoiseAccelerometer. The colored noise expression is as below:
![image](https://github.com/groops-devs/groops/assets/95745305/6abb9123-8b36-4c61-a8f3-f07016543b9a)

For this purpose, I set the parameters as follows:
![image](https://github.com/groops-devs/groops/assets/95745305/c957e70a-828f-46e1-80f8-4a9bb7a2ff0a)

However, there was a warning during operation:
![image](https://github.com/groops-devs/groops/assets/95745305/85f2a909-7cf8-40d4-8364-971521679cb3)

And the acc along data is all 'nan'.
![image](https://github.com/groops-devs/groops/assets/95745305/a6498b59-5ca3-4982-82b6-811547a34d13)

How could I solve this problem?

Thank you in advance for your feedback and help.
With Kind Regards,
Kyrie714

**Comment by tmayerguerr** [comment, marked as answer]
Hi Kyrie714,

The problem is that your PSD expression is also evaluated for the 0 Hz frequency:
`"Warning: PSD at frequency 0 Hz is nan"`
This might be solved by extending your expression by a condition like:
`if(freq==0, 0, your expr)`

best regards
Torsten

**Reply by Kyrie714** [reply]
Thank you very much! It helps!
